### ISSUE TITLE
Bootstrap git-flow project with Flux v2

### ISSUE THREAD

**Issue by devigned** [Azure/azure-capi-cli-extension#18]
Getting started with git-flow can be daunting. We should help folks get started with CAPI / CAPZ by setting them up with a batteries included git-flow project(s) using Flux v2.

Perhaps, we can start with a proposal for how we can blend https://toolkit.fluxcd.io/get-started/ with `az capi init`.

I'm imagining a command to init a repo, setup a `./clusters/my-mgmt-cluster` path with a kustomize set of CAPI / CAPZ resources, deploy a management cluster with Flux components and point to the new repo.

**Comment by eedorenko** [comment]
Proposal:

The goal is to enrich **az capi** extension with the GitOps bootstrapping/configuring functionality. Here is [an example of a GitOps setup with CAPI/CAPZ and Flux](https://eugene-fedor.medium.com/provision-a-fleet-of-aks-clusters-in-gitops-fashion-f7776693cc52).

To automate the GitOps configuration, the extension should be able to do the following:
  - Bootstrap Flux on the management cluster
  - Create and maintain on the management cluster Flux resources (GitRepository, Kustomization) to deliver CAPI resources from a Git repository to the cluster

It is proposed to:
- Add the following keys to **az capi management create/update**:
  - flux-repo ([Flux bootstrap repository](https://fluxcd.io/docs/installation/#bootstrap)  https://username:password@<host>/<org>/<repo>
  - flux-branch (Default "main") 
  - flux-path (Default "flux-system") 
  
  The command bootstraps Flux on the management cluster. If provided repo, branch and path don't exist, it will create them.  

- Introduce **az capi fleet** set of commands:
  - **az capi fleet add** with the following keys:
    - repo (Git repo with cluster yamls. https://username:password@<host>/<org>/<repo>)
    - branch (Branch in the Git repo with cluster yamls. Default "main")
    - name (Name of cluster group represented by a folder in the fleet repo. Default "capi")
    - path (Path in the Git repo with cluster yamls. Default =name)
    - namespace (Namespace where workload cluster yamls should be reconciled)
    The command creates in the management cluster a pair of labeled GitRepository, Kustomization Flux resources to pull workload cluster yamls from a Git repository folder that represents a fleet/group of clusters. If provided repo, branch and path don't exist, it will create them. The command creates a PR with the GitRepository and Kustomization yamls to store them in the Git repo.

    It would also be beneficial to add similar keys (e.g. fleet-repo, fleet-name, ...) to **az capi management create/update** so the whole initial setup can be created with a single command. In most cases there will be a single fleet of workload clusters.

  - **az capi fleet delete** to delete a specific fleet. It will create a PR with the deletion of the fleet folder. When the PR is merged, Flux will delete all reconciled workload clusters of this fleet along with the corresponding GitRepository and Kustomization.
  - **az capi fleet list** to list defined workload cluster fleets. The command analyzes labeled pairs of GitRepository/Kustomization that represent the cluster fleets.
  - **az capi fleet show** to show the details of the specific fleet including a list of fleet clusters 

- Add a key to **az capi create/update/delete**:
  - fleet (Name of the workload cluster fleet. The command creates a PR to the fleet folder to add/update/remove a cluster. Once the PR is merged, Flux will deliver changes to the management cluster).

**Comment by devigned** [comment]
@eedorenko I was really hoping the new project experience to be something like `az capi init ${projectName}` where a local directory is created with an initial project structure (management cluster, first workload cluster, any github actions, etc), git init'd in that directory, and perhaps, automate the creation of the remote GitHub repository, then adding a named remote to the local git.

This getting started experience would be in the spirit of https://devcenter.heroku.com/articles/creating-apps or https://guides.rubyonrails.org/command_line.html#rails-new where a functional project is bootstrapped and ready to be extended by the user. Getting started with GitFlow is tough for folks. I'm hoping we can make is supremely simple.

**Comment by eedorenko** [comment]
@devigned, yes,  ```az capi init``` may be added on top of what is proposed above to combine all steps together. Roughly, it will perform the following:
```
# 1. Init repo locally and push it remotely
git init $projectName}
cd ${projectName}
gh repo create

# 2. Initialize management cluster and bootsrap Flux 
# It will add "flux-system" folder to the "main" branch of the new repo
az capi management create

# 3. Add a cluster fleet
# It will add a fleet folder "capi" in the "main" branch with GitRepository, Kustomization Flux resources
# to reconcile workload clusters in the fleet 
az capi fleet add

# 4. Add a cluster to the fleet
# It will add cluster resources to the fleet folder in the repo,
# so it will be reconciled by Flux and the cluster will be created by CAPI/CAPZ 
az capi create --fleet=capi --name={clusterName} --location ... all needed parameters for "az capi create" 

```     

That said, ```az capi init``` will need configurations/parameters for all 4 steps, such as:
 - GH credentials
 - Management cluster configs (e.g. Kubeconfig, Subscription)
 - Fleet name and namespace, if not default
 - Workload cluster cluster configs 

All these parameters may be set through the command keys and/or they can be requested from user "on the fly" as it happens now for some of them. 

This is a matter of preference of using such an "init-all-stuff" command. I would, personally, prefer to go step by step rather than provide all necessary configurations/parameters, shoot and hope I didn't make a mistake. But I definitely see the use-case for such command when people want to use it as a starting point or just get a taste of what it is all about.     

In any scenario, after the initial setup is done (with one or multiple steps), the Day 2 will come. This is what my proposal is focused on, to help with daily tasks to maintain cluster fleets, workload and management clusters. The workload clusters are to be added/updated/removed. The management clusters is also not a "once setup" thing, especially when they are short-living/ephemeral. 

In summary, I think the proposal is extended with the ```az capi init``` command as described above in addition to [the original one](https://github.com/Azure/azure-capi-cli-extension/issues/18#issuecomment-853194307).
